Framework/Geometry/test/BeamlineRayTracerTest.h

```diff
@@ -0,0 +1,615 @@
+#ifndef BEAMLINERAYTRACERTEST_H_
+#define BEAMLINERAYTRACERTEST_H_
+
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidGeometry/Instrument/InstrumentDefinitionParser.h"
+#include "MantidGeometry/Instrument/InstrumentVisitor.h"
+#include "MantidGeometry/Instrument/RectangularDetector.h"
+#include "MantidGeometry/Objects/BeamlineRayTracer.h"
+#include "MantidGeometry/Objects/InstrumentRayTracer.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidKernel/ProgressText.h"
+#include "MantidKernel/Strings.h"
+#include "MantidTestHelpers/ComponentCreationHelper.h"
+
+#include <boost/make_shared.hpp>
+#include <cxxtest/TestSuite.h>
+
+using namespace Mantid::Geometry;
+using Mantid::Kernel::ConfigService;
+using Mantid::Kernel::ProgressText;
+using Mantid::Kernel::V3D;
+using namespace Mantid::Kernel::Strings;
+using namespace ComponentCreationHelper;
+namespace RayTracer = Mantid::Geometry::BeamlineRayTracer;
+
+class BeamlineRayTracerTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static BeamlineRayTracerTest *createSuite() {
+    return new BeamlineRayTracerTest();
+  }
+  static void destroySuite(BeamlineRayTracerTest *suite) { delete suite; }
+
+  BeamlineRayTracerTest() {
+    // Start logging framework
+    Mantid::Kernel::ConfigService::Instance();
+  }
+
+  void
+  test_that_a_trace_for_a_ray_that_intersects_many_components_gives_these_components_as_a_result() {
+    // Create the test objects to use
+    create_instrument_and_componentInfo();
+
+    // Vector for a ray
+    V3D testDir(0., 0., 1);
+
+    // Do a trace and store the results
+    Links results = RayTracer::traceFromSource(testDir, *m_compInfo);
+
+    // Check size
+    TS_ASSERT_EQUALS(results.size(), 2);
```

Comment:
It's not directly clear from the test setup that we expect it to apss through 2 components. Maybe we could add some description about that. i.e. what components are relevant from `create_instrument_and_componentInfo`